### PR TITLE
Add vehicle drop off restrictions

### PR DESCRIPTION
**Pull request by josee-sabourin**
#### **What problem does your proposal solve? Please begin with the relevant issue number. If there is no existing issue, please also describe alternative solutions you have considered.**
As discussed in issue #270, there is no way to describe how a vehicle should be returned at the end of a trip.  

#### **What is the proposal?**
This proposal adds a `return_type` field to `vehicle_types.json` that allows operators to define if the vehicle is `free_floating` (vehicle must be returned anywhere within the service area), `roundtrip` (vehicle must be returned at original rental station), or `station` (vehicle must be returned to any station within the service area). The field is modelled as an array to account for hybrid vehicles that have multiple ways of being returned. 

Additionally, to account for edge cases where a vehicle must be returned to a particular station that is not the original station, a `home_station` field has been added to `free_bike_status.json`.  

Read the full extension proposal [here](http://bit.ly/GBFS-DropOffs-Proposal).

#### **Is this a breaking change?**
- [ ] Yes 
- [x] No
- [ ] Unsure

#### **Which files are affected by this change?**
- vehicle_types.json
- free_bike_status.json

### CODE REVIEW

File: gbfs.md [hunk 1]
```diff
@@ -395,6 +395,7 @@ Field Name | REQUIRED | Type | Defines
...
 \- `max_range_meters` | Conditionally REQUIRED | Non-negative float | If the vehicle has a motor (as indicated by having a value other than `human` in the `propulsion_type` field), this field is REQUIRED. This represents the furthest distance in meters that the vehicle can travel without recharging or refueling when it has the maximum amount of energy potential (for example, a full battery or full tank of gas).
 \- `name` | OPTIONAL | String | The public name of this vehicle type.
+\- `return_type` | OPTIONAL | Array | The conditions for returning the vehicle at the end of the trip. For vehicles that have more than one return option, include all applicable methods in the array. <br /><br />Current valid values are:<br /><ul><li>`free_floating` _(The vehicle can be returned anywhere permitted within the service area)_</li><li>`roundtrip` _(The vehicle must be returned to the initial rental station)_</li><li>`station` _(The vehicle must be returned to any station within the service area - note that a specific station can be defined in [free_bike_status.json](#free_bike_status.json) using `home_station`)_
```
Comment (kanagy) [inline gbfs.md]: Hello! Could you add:
- That free_floating vehicles return is subject to geofencing zones if such feed is defined
- That rountrip value cannot be used in combination with free_floating bikes. It doesn't make sense.
- I would rename station to any_station, and rountrip to roundtrip_station. And would add any_station cannot be used in combination with roundrtip_station (the two don't make sense together). Renaming them makes it clear that the two are exclusive.

I also left a comment on the proposal doc about home_station systems. I would like to see some more examples of such system.

**Review by kanagy — COMMENTED** [review]
Thanks for the proposal!

File: gbfs.md (hunk 1 above)
Comment (josee-sabourin) [reply gbfs.md]: Thanks for your comments Kristian, I will make those changes! I will answer your question about home_station in the proposal document.

**Comment by kanagy** [comment]
Hello! I'd love to hear about aggregators / transportation apps on [needs assessment](https://docs.google.com/document/d/1h3tonIpbxpgvidQsNVQVdzud3wjFdbVyXLr0ITh-qZ0/edit#) in #270, specifically (sorry to be late):
- Are there any aggregators currently that already support hybrid systems (as described in the doc)? If so how are they currently represented in the feeds and does this proposal work for you? 
- Is there an interest to support A to A systems, e.g. bike rental shops? This is a curious use case for transportation apps (built for getting from A to B); it would mean surfacing these it as a viable option for an A to B routes but then somehow cautioning that a user needs to get back to A eventually.

**Comment by viestat** [comment]
Hello!

Is there any intention to merge this PR?
This PR along with https://github.com/NABSA/gbfs/pull/349 are needed in order to resolve the need outlaid in https://github.com/NABSA/gbfs/issues/332.

Thanks!

**Comment by mplsmitch** [comment]
@viestat  In order to merge this PR it needs to be voted on. We'll be calling a vote as soon as we've got the necessary support lined up. In order to pass we need at least 3 votes including one from a data producer and one from a data consumer. Would you be willing to support this as the data producer?

Full governance/change process is [here](https://github.com/NABSA/gbfs#governance--overview-of-the-change-process)

**Comment by viestat** [comment]
> Would you be willing to support this as the data producer?

@mplsmitch thank you for clarifying, I am willing to support it.

**Comment by josee-sabourin** [comment]
I hereby call a vote on this proposal. Voting will be open for 10 full calendar days until 11:59PM UTC on September 1st, 2021.
Please vote for or against the proposal, and include the organization for which you are voting in your comment.
Please note if you can commit to implementing the proposal.

EDIT: Tagging for previous engagement in issue/google docs - @hbruch @leonardehrenfried @flaktack @viestat

**Comment by kanagy** [comment]
Google Maps supports this proposal. We don't intend on immediately supporting it, but think this will be useful in the future to model increasingly hybrid systems.

I think minor changes would be useful in this PR:
1) Renaming `return_types` to `return_constraints` seems more accurate. It makes it clear that these are optional constraints that when defined routing apps should respect.
2) Optionally, adding `any_charging_station` as a constraint after #340 PR is in.

**Comment by testower** [comment]
Entur supports this proposal and will implement it in our aggregation service.

**Comment by nbdh** [comment]
Nextbike supports this proposal. As such limitations are rather scarce in our systems publishing this field probably won't have priority for us.

**Comment by NicolasFrasie** [comment]
Communauto supports this proposal. It is required to specify if a vehicle is round-trip for GBFS to be able to model round trip carsharing (such as Communauto and Modo in Canada, Zipcar and Enterprise CarShare in US, Citiz in France, etc.). 

This PR is a prerequisite for PR #350 which aims to extend the current GBFS specification to allow car sharing support (roundtrip like Zipcar, free-floating like GIG CarShare and station-to-station like BlueLA)

**Comment by viestat** [comment]
Dott supports this proposal, however we will not implement this until https://github.com/NABSA/gbfs/pull/349 (or any alternative proposal that fully addresses https://github.com/NABSA/gbfs/issues/332) is merged.

**Comment by Miryad3108** [comment]
The French national access point to transport data [**transport.data.gouv.fr]**(https://transport.data.gouv.fr/) supports this proposal which has been integrated into the PR [#350](https://github.com/NABSA/gbfs/pull/350)

**Comment by Miryad3108** [comment]
The French national access point to transport data [transport.data.gouv.fr ](https://transport.data.gouv.fr/)supports this proposal which has been integrated into the PR [#350](https://github.com/NABSA/gbfs/pull/350)

**Comment by evansiroky** [comment]
IBI Group supports this proposal, although we recommend renaming `free_bike_status.json#home_station` to `free_bike_status.json#home_station_id` to try to follow a naming convention that shows which fields have IDs as values.

**Comment by kanagy** [comment]
+1 for renaming to `home_station` to `home_station_id` for consistency with other fields

**Comment by cait32** [comment]
BCycle supports this proposal, but will not implement it immediately.

**Comment by ncancelliere** [comment]
Spin supports this proposal. We don’t plan on immediately supporting it.

**Comment by ezmckinn** [comment]
Superpedestrian supports this proposal, but like @viestat noted, would not implement until [#349](https://github.com/NABSA/gbfs/pull/349) is merged.

**Comment by josee-sabourin** [comment]
This vote is now closed, and it passes! 

Votes in favour: 
Google Maps (consumer)
Entur (consumer)
Nextbike (producer)
Communauto (producer)
Dott (producer)
Transport.data.gouv.fr (consumer)
IBI Group (consumer)
BCycle (producer)
Spin (producer)
Superpedestrian (producer)

Regarding @kanagy's and @evansiroky's comments: We will make the semantic changes once we merge into v2.3-RC. Regarding any_charging_station, we will open an issue to determine the need for this field.

**Pull request merged**